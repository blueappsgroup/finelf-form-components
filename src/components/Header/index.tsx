import React from 'react'
import styled from 'styled-components'
import { device } from '../../consts/sizes'
import { HeaderProps } from '../../types'

type StyledProps = {
  theme: {
    headerTextColor: string
    headerBorderColor: string
    subheaderTextColor: string
  }
  withBorder?: boolean
}

// prettier-ignore
const StyledMainHeader = styled.h2<StyledProps>`
  font-weight: 600;
  font-size: 25px;
  line-height: 24px;
  text-align: center;
  margin: 0px 0px 20px 0px;
  padding-bottom: 28px;
  position: relative;
  display: flex;
  justify-content: center;
  color: ${(props: StyledProps): string => props.theme.headerTextColor};
  position: relative;
  
  @media ${device.tablet} {
    margin: 0px 0px 30px 0px;
  }

  &::after {
    width: 33%;
    content: '';
    position: absolute;
    bottom: 0px;
    border: ${(props: StyledProps): string => props.withBorder ? `3px solid ${props.theme.headerBorderColor}` : 'none'};
    margin: 0 auto;
  }
`

const StyledSubHeader = styled.h5<StyledProps>`
  color: ${(props: StyledProps): string => props.theme.subheaderTextColor};
  font-style: normal;
  font-weight: 600;
  font-size: 16px;
  line-height: 18px;
  margin: 10px 0px 20px 0px;
`

const Header: React.FC<HeaderProps> = ({ text, type = 'main', withBorder }) =>
  type === 'main' ? (
    <StyledMainHeader withBorder={withBorder}>{text}</StyledMainHeader>
  ) : (
    <StyledSubHeader>{text}</StyledSubHeader>
  )

export default Header
