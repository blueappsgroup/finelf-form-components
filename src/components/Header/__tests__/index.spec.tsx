import { render } from '@testing-library/react'
import * as React from 'react'
import Header from '../'

describe('<Header />', () => {
  it('main header matches snapshot', () => {
    const documentBody = render(<Header text="Header" />)
    const { baseElement } = documentBody

    expect(documentBody.getByText('Header')).toBeTruthy()
    expect(baseElement).toMatchSnapshot()
  })

  it('main header with border matches snapshot', () => {
    const documentBody = render(<Header text="Header" withBorder />)
    const { baseElement } = documentBody

    expect(documentBody.getByText('Header')).toBeTruthy()
    expect(baseElement).toMatchSnapshot()
  })

  it('subheader matches snapshot', () => {
    const documentBody = render(<Header text="Header" type="subheader" />)
    const { baseElement } = documentBody

    expect(documentBody.getByText('Header')).toBeTruthy()
    expect(baseElement).toMatchSnapshot()
  })
})
