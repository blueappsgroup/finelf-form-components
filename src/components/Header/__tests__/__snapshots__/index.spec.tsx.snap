// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> main header matches snapshot 1`] = `
<body>
  <div>
    <h2
      class="sc-bdnylx dIsRVf"
    >
      Header
    </h2>
  </div>
</body>
`;

exports[`<Header /> main header with border matches snapshot 1`] = `
<body>
  <div>
    <h2
      class="sc-bdnylx ea-DsHd"
    >
      Header
    </h2>
  </div>
</body>
`;

exports[`<Header /> subheader matches snapshot 1`] = `
<body>
  <div>
    <h5
      class="sc-gtssRu jcgMrl"
    >
      Header
    </h5>
  </div>
</body>
`;
